src/components/Image/Image.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import styles from './Image.module.scss';
+
+type Props = {
+  className?: string;
+  src?: string;
+  fallbackSrc?: string;
+  onLoad?: () => void;
+  alt?: string;
+};
+
+const Image = ({ className, src, fallbackSrc, onLoad, alt = '' }: Props) => {
+  const [imgSrc, setImgSrc] = useState(src);
+
+  const handleLoad = () => {
+    if (onLoad) onLoad();
+  };
+
+  const handleError = () => {
+    if (fallbackSrc && fallbackSrc !== src) {
+      setImgSrc(fallbackSrc);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
I think this should use a useLayoutEffect because it runs before the browser has a chance to paint the screen.

Reply:
I don't think it is needed here since we're not relying on the src to be set directly. 

https://kentcdodds.com/blog/useeffect-vs-uselayouteffect